src/Drupal/DrupalExtension/Context/DrupalContext.php

```diff
@@ -183,32 +183,134 @@ public function getTableRow(Element $element, $search)
         throw new \Exception(sprintf('Failed to find a row containing "%s" on the page %s', $search, $this->getSession()->getCurrentUrl()));
     }
 
+  /**
+   * Retrieve table rows containing specified text from a given element.
+   *
+   * @param Element $element
+   *   \Behat\Mink\Element\Element object.
+   * @param string $search
+   *   The text to search for in the table row.
+   *
+   * @return array of \Behat\Mink\Element\NodeElement
+   *
+   * @throws \RuntimeException
+   */
+  public function getTableRows(Element $element, $search): array {
+    $rows = $element->findAll('css', 'tr');
+    if (empty($rows)) {
+      throw new \RuntimeException(sprintf('No rows found on the page %s', $this->getSession()->getCurrentUrl()));
+    }
+    $found_rows = [];
+    foreach ($rows as $row) {
+      if (str_contains($row->getText(), $search) !== false) {
+        $found_rows += $row;
+      }
+    }
+    if (count($found_rows) === 0) {
+      throw new \RuntimeException(sprintf('Failed to find a row containing "%s" on the page %s', $search, $this->getSession()->getCurrentUrl()));
+    }
+    return $found_rows;
+  }
+
+  /**
+   * See if the element has a table.
+   *
+   * @param Element $element
+   *    \Behat\Mink\Element\Element object.
+   *
+   * @return bool
+   *   True if the element has a table.
+   */
+  private function hasTable(Element $element): bool {
+    $rows = $element->findAll('css', 'tr');
+    return empty($rows);
+  }
+
   /**
    * Find text in a table row containing given text.
    *
    * @Then I should see (the text ):text in the :rowText row
    */
-    public function assertTextInTableRow($text, $rowText)
-    {
-        $row = $this->getTableRow($this->getSession()->getPage(), $rowText);
-        if (strpos($row->getText(), $text) === false) {
-            throw new \Exception(sprintf('Found a row containing "%s", but it did not contain the text "%s".', $rowText, $text));
+    public function assertTextInTableRow($text, $rowText) {
+      $rows = $this->getTableRows($this->getSession()->getPage(), $rowText);
+      foreach ($rows as $row) {
+        if (str_contains($row->getText(), $text) !== false) {
+          return;
         }
+      }
+      throw new \Exception(sprintf('Found a row containing "%s", but it did not contain the text "%s".', $rowText, $text));
+    }
+
+  /**
+   * Find text in all table rows containing given text.
+   *
+   * @Then I should see (the text ):text in all the :rowText rows
+   */
+  public function assertTextInTableRows($text, $rowText) {
+    $rows = $this->getTableRows($this->getSession()->getPage(), $rowText);
+    foreach ($rows as $row) {
+      if (str_contains($row->getText(), $text) !== false) {
+        throw new \Exception(sprintf('Found a row containing "%s", but it did not contain the text "%s".', $rowText, $text));
+      }
     }
```

Comment:
@kumikoono Let's use a guard clause:

```suggestion
    foreach ($rows ?: [] as $row) {
      if (str_contains($row->getText(), $text) === FALSE) {
        continue;
      }
      throw new \Exception(sprintf('Found a row containing "%s", but it did not contain the text "%s".', $rowText, $text));
    }
```